### ISSUE TITLE
Add static code analysis

### ISSUE THREAD

**Issue by sils1297** [mapsme/omim#2906]
Hi,

I'm a very happy user of this app and coincidentally I'm the author of a static code analysis framework coala and the service connected to it GitMate which can automatically review your pull requests (gitmate.io).

Apparently you don't have static code analysis yet, I can offer setting it up for you (we're also interested in trying the setup on some more sophisticated projects so this should be a win win). It can run on all languages and your docs as well without you having to bother with lots of different tools. It's easily integratable into any CI system or GitMate can be used for free (it'll just comment about newly introduced issues in the GitHub diff).

coala is AGPL and runs totally without any proprietary piece of software.

Is that interesting for you? What all would you like to have analyzed?

**Comment by deathbaba** [comment]
Hi, can you please analyze our omim repository? Let's check the result :)

**Comment by sils1297** [comment]
Alright, let's see what we have here

/me runs cloc.

**Comment by blobgo** [comment]
Just FYI there are many other code analyzers: E.g. https://codacy.com/ and [many other](https://github.com/integrations/feature/code)...

**Comment by sils1297** [comment]
Yes, there are lots of alternatives. If you want the *best* analysis for C your best bet is probably coverity scan for open source. (Good luck setting it up though and it's proprietary.) All the other platforms (codacy, scrutinizer, ...) are usually also proprietary and don't offer a real deal breaker.

You can either go for the good results (coverity etc.) or have something simple (codacy, gitmate, etc.) and you can choose between free and nonfree software. (HoundCI is a free alternative that does style checking only though. GitMate is coala based and all algorithms are free.) If you want to do analysis on multiple languages at once you'll have a hard time finding something that beats coala in simplicity, you can set everything up in one config and rule a whole fleet of open source tools and our native algorithms at once.

I'm of course biased towards GitMate and coala but hope the overview helps a bit. We're spending a bit of time on the competition and obviously they didn't convince us :P
